src/py/migraphx_py.cpp

```diff
@@ -261,6 +262,20 @@ MIGRAPHX_PYBIND11_MODULE(migraphx, m)
         .def("__ne__", std::not_equal_to<migraphx::shape>{})
         .def("__repr__", [](const migraphx::shape& s) { return migraphx::to_string(s); });
 
+    py::enum_<migraphx::shape::type_t>(shape_cls, "type_t")
+        .value("bool_type", migraphx::shape::type_t::bool_type)
+        .value("half_type", migraphx::shape::type_t::half_type)
+        .value("float_type", migraphx::shape::type_t::float_type)
+        .value("double_type", migraphx::shape::type_t::double_type)
+        .value("uint8_type", migraphx::shape::type_t::uint8_type)
+        .value("int8_type", migraphx::shape::type_t::int8_type)
+        .value("uint16_type", migraphx::shape::type_t::uint16_type)
+        .value("int16_type", migraphx::shape::type_t::int16_type)
+        .value("int32_type", migraphx::shape::type_t::int32_type)
+        .value("int64_type", migraphx::shape::type_t::int64_type)
+        .value("uint32_type", migraphx::shape::type_t::uint32_type)
+        .value("uint64_type", migraphx::shape::type_t::uint64_type);
```

Comment:
This should be done using `MIGRAPHX_SHAPE_VISIT_TYPES`. I think it would be something like:

```cpp
#define MIGRAPHX_PYTHON_GENERATE_SHAPE_ENUM(x, t) \
    .value(#x, migraphx::shape::type_t::t);

py::enum_<migraphx::shape::type_t>(shape_cls, "type_t")
    MIGRAPHX_SHAPE_VISIT_TYPES(MIGRAPHX_PYTHON_GENERATE_SHAPE_ENUM)
```